### PR TITLE
Handle geo location failures gracefully

### PR DESCRIPTION
**Pull request by mapkyca**
## Here's what I fixed or added:

* Debug to geo location checkin code
* Error handler now displays error messages
* When location fails, we initialise the map to a default location and allow manual entry

## Here's why I did it:

Location failures would give an infinite loop

### CODE REVIEW

File: IdnoPlugins/Checkin/checkin.js [hunk 1]
```diff
@@ -1,67 +1,70 @@
...
+	    longitude = position.coords.longitude;
+    queryLocation(latitude, longitude, initMap.bind(this, latitude, longitude));
+}
 
-    function handleErrorFromNavigator() {
-    }
+function handleErrorFromNavigator(err) {
+    addErrorMessage(err.message);
+    //$('#geoplaceholder').html('<p>Could not find your location: '+err.message +'</p>');
+    initMap("51.478791", "-0.01068"); // Allow failures to allow manual entry. Default to the center of the universe.
```
Comment (kylewm) [inline IdnoPlugins/Checkin/checkin.js]: ha! :+1:

File: IdnoPlugins/Checkin/checkin.js [hunk 2]
```diff
@@ -1,67 +1,70 @@
 var CheckinMap, CheckinMarker;
-$(function () {
```
Comment (kylewm) [inline IdnoPlugins/Checkin/checkin.js]: why move this down? did these functions need to be in the global scope?

Comment (mapkyca) [inline IdnoPlugins/Checkin/checkin.js]: Probably not, I'll change that back.

**Comment by kylewm** [comment]
Lgtm!

**Pull request merged**